spec/lib/services/pollers-api-service-spec.js

```diff
@@ -0,0 +1,251 @@
+            var mockPoller = {"message":"NotFoundError: Could not find workitem with identifier"};
+            waterline.workitems.find.rejects(mockPoller);
+            return pollerService.getPollers().should.eventually.be.rejectedWith(mockPoller);
+        });
+
+    });
+
+    describe("getPollersById", function() {
+        it("should expose the appropriate methods", function() {
+            pollerService.should.have.property("getPollersById")
+            .that.is.a("function").with.length(1);
+        });
+
+        it("Run getPollersById", function() {
+            var mockPoller = [{
+                id: "1234",
+                name: "Pollers.IPMI",
+                config: {}
+            },
+            {
+                id: "4532",
+                name: "Pollers.TEST",
+                config: {}
+            }];
+            waterline.workitems.needByIdentifier.withArgs({"id": "4532"}).resolves(mockPoller[0]);
+            waterline.workitems.needByIdentifier.withArgs({"id": "1234"}).resolves(mockPoller[1]);
+            Promise.all(["4532", "1234"].map(function(id, index) {
+                return pollerService.getPollersById(index)})).then(function (pollers) {
+                    expect(pollers).to.deep.equal(mockPoller[index]);
+                });
+         });
+
+        it("should return error if specific poller info is not found", function () {
+            var mockPoller = {"message":"NotFoundError: Could not find workitem with identifier"};
+            waterline.workitems.needByIdentifier.rejects(mockPoller);
+            return pollerService.getPollersById().should.eventually.be.rejectedWith(mockPoller);
+        });
+
+    });
+
+    describe("getPollerLib", function() {
+        it("should expose the appropriate methods", function() {
+            pollerService.should.have.property("getPollerLib")
+            .that.is.a("function").with.length(0);
+        });
+
+    });
+
+    describe("getPollerLibById", function() {
+        it("should expose the appropriate methods", function() {
+            pollerService.should.have.property("getPollerLibById")
+            .that.is.a("function").with.length(1);
+        });
+
+    });
+
+    describe("postPollers", function() {
+        it("should expose the appropriate methods", function() {
+            pollerService.should.have.property("postPollers")
+            .that.is.a("function").with.length(1);
+        });
+
+        it("Run postPollers", function() {
+            var mockPoller = [{
+                id: "4532",
+                name: "Pollers.IPMI",
+                config: {}
+            }];
+            waterline.workitems.create.resolves(mockPoller);
+            return pollerService.postPollers().then(function (pollers) {
+                expect(pollers).to.deep.equal(mockPoller);
+            });
+        });
+
+        it("Throws error when  postPollers runs with invalid input", function() {
+            var mockPoller = {
```

Comment:
@srinia6 I think we want to make this an error object: `var mockPollerError = new Errors.ValidationError('Validation errors')`.